### PR TITLE
[FIX JENKINS-31015] Prevent InvalidPathException in FileVF

### PR DESCRIPTION
**Pull request by daniel-beck**

### CODE REVIEW

**Comment by daniel-beck** [comment]
Only used for PR build for now.

**Comment by oleg-nenashev** [comment]
Looks good to me BTW

File: core/src/main/java/hudson/model/DirectoryBrowserSupport.java [hunk 1]
```diff
@@ -183,9 +184,13 @@ private void serveFile(StaplerRequest req, StaplerResponse rsp, VirtualFile root
...
                 // Treat * and ? as wildcard unless they match a literal filename
-                if((pathElement.contains("?") || pathElement.contains("*"))
-                        && inBase && !root.child((_base.length() > 0 ? _base + "/" : "") + pathElement).exists())
-                    inBase = false;
+                try {
+                    if((pathElement.contains("?") || pathElement.contains("*"))
+                            && inBase && !root.child((_base.length() > 0 ? _base + "/" : "") + pathElement).exists())
+                        inBase = false;
+                } catch (InvalidPathException e) {
+                    // * is an illegal filename component on some systems, see JENKINS-31015 -- ignore
```
Comment (jtnord) [inline core/src/main/java/hudson/model/DirectoryBrowserSupport.java]: should this not return an error page or at least break the while loop or something here?

**Comment by daniel-beck** [comment]
Catching the cause at the source, supersedes original e18d2d1156d80ad19c51143495776e415cb3d29e.

**Comment by daniel-beck** [comment]
Got confirmation via Jira that the new PR build fixes the issue. Anyone objecting to merging this?

**Comment by oleg-nenashev** [comment]
Nope, :+1: for merge

**Pull request merged**